File: android/app/src/main/java/com/preuvely/app/data/models/Banner.kt
package com.preuvely.app.data.models

import androidx.compose.ui.graphics.Color
import com.google.gson.annotations.SerializedName

enum class BannerLinkType(val value: String) {
    @SerializedName("none") NONE("none"),
    @SerializedName("store") STORE("store"),
    @SerializedName("category") CATEGORY("category"),
    @SerializedName("url") URL("url")
}

data class Banner(
    @SerializedName("id") val id: Int,
    @SerializedName("title") val title: String?,
    @SerializedName("subtitle") val subtitle: String?,
    @SerializedName("image_url") val imageUrl: String,
    @SerializedName("background_color") val backgroundColor: String,
    @SerializedName("link_type") val linkType: BannerLinkType,
    @SerializedName("link_value") val linkValue: String?
) {
    val color: Color
        get() = try {
            Color(android.graphics.Color.parseColor(backgroundColor))
        } catch (e: Exception) {
            Color(0xFF007359) // Default to primary green
        }

    val hasLink: Boolean
        get() = linkType != BannerLinkType.NONE && !linkValue.isNullOrBlank()
}

data class BannerListResponse(
    @SerializedName("data") val data: List<Banner>
)

File: ios/Preuvely/Preuvely/Models/Banner.swift
import Foundation
import SwiftUI

// MARK: - Banner Model

struct Banner: Codable, Identifiable, Equatable {
    let id: Int
    let title: String?
    let subtitle: String?
    let imageUrl: String
    let backgroundColor: String
    let linkType: LinkType
    let linkValue: String?

    enum LinkType: String, Codable {
        case none
        case store
        case category
        case url
    }

    // Note: Using JSONDecoder with .convertFromSnakeCase strategy
    // No explicit CodingKeys needed - snake_case is auto-converted to camelCase

    /// Convert hex color string to SwiftUI Color
    var color: Color {
        Color(hex: backgroundColor) ?? .primaryGreen
    }

    /// Check if banner has a valid link
    var hasLink: Bool {
        linkType != .none && linkValue != nil && !linkValue!.isEmpty
    }
}

// MARK: - Preview Data

#if DEBUG
extension Banner {
    static let samples: [Banner] = []
}
#endif

// MARK: - Color Extension for Hex

extension Color {
    init?(hex: String) {
        var hexSanitized = hex.trimmingCharacters(in: .whitespacesAndNewlines)
        hexSanitized = hexSanitized.replacingOccurrences(of: "#", with: "")

        var rgb: UInt64 = 0

        guard Scanner(string: hexSanitized).scanHexInt64(&rgb) else {
            return nil
        }

        let r = Double((rgb & 0xFF0000) >> 16) / 255.0
        let g = Double((rgb & 0x00FF00) >> 8) / 255.0
        let b = Double(rgb & 0x0000FF) / 255.0

        self.init(red: r, green: g, blue: b)
    }
}
